Cmdline/Action/Search.cs

```diff
@@ -16,60 +16,150 @@ public Search(IUser user)
+            // options.search_term = options.search_term ?? string.Empty;
 
             // Show how many matches we have.
-            user.RaiseMessage("Found " + matching_mods.Count().ToString() + " mods matching \"" + options.search_term + "\".");
+            if (options.all && !String.IsNullOrWhiteSpace(options.author_term))
+            {
+                user.RaiseMessage("Found {0} compatible and {1} incompatible mods matching \"{2}\" by \"{3}\".",
+                    matching_compatible.Count().ToString(),
+                    matching_incompatible.Count().ToString(),
+                    options.search_term,
+                    options.author_term);
+            }
+            else if (options.all && String.IsNullOrWhiteSpace(options.author_term))
+            {
+                user.RaiseMessage("Found {0} compatible and {1} incompatible mods matching \"{2}\".",
+                    matching_compatible.Count().ToString(),
+                    matching_incompatible.Count().ToString(),
+                    options.search_term);
+            }
+            else if (!options.all && !String.IsNullOrWhiteSpace(options.author_term))
+            {
+                user.RaiseMessage("Found {0} compatible mods matching \"{1}\" by \"{2}\".",
+                    matching_compatible.Count().ToString(),
+                    options.search_term,
+                    options.author_term);
+            }
+            else if (!options.all && String.IsNullOrWhiteSpace(options.author_term))
+            {
+                user.RaiseMessage("Found {0} compatible mods matching \"{1}\".",
+                    matching_compatible.Count().ToString(),
+                    options.search_term);
+            }
 
             // Present the results.
-            if (!matching_mods.Any())
+            if (!matching_compatible.Any() && (!options.all || !matching_incompatible.Any()))
             {
                 return Exit.OK;
             }
 
-            // Print each mod on a separate line.
-            foreach (CkanModule mod in matching_mods)
+            if (options.detail)
+            {
+                user.RaiseMessage("Matching compatible mods:");
+                foreach (CkanModule mod in matching_compatible)
+                {
+                    user.RaiseMessage("* {0} ({1}) - {2} by {3} - {4}",
+                        mod.identifier,
+                        mod.version,
+                        mod.name,
+                        mod.author == null ? "N/A" : String.Join(", ", mod.author),
+                        mod.@abstract);
+                }
+
+                if (matching_incompatible.Any())
+                {
+                    user.RaiseMessage("Matching incompatible mods:");
+                    foreach (CkanModule mod in matching_incompatible)
+                    {
+                        Registry.GetMinMaxVersions(new List<CkanModule> { mod } , out _, out _, out var minKsp, out var maxKsp);
```

Comment:
Oh, I didn't know you could discard an `out` param with "_". Cool; there are some places where I'll want to use that (mostly other spots that call `GetMinMaxVersions`).

Reply:
Yep, searched for it because I wanted to know if it's possible. There are quite a few answers on stackoverflow mentioning them if you search for `c# discard out parameter` or so.